Infrastructure/ARM/Deploy.json

```diff
@@ -0,0 +1,299 @@
+            "apiVersion": "2022-03-01",
+            "kind": "functionapp",
+            "location": "[variables('location')]",
+            "sku": {
+            "name": "Y1"
+            },
+            "properties": {
+            "reserved": false
+            }
+        },
+        // Azure function app
+        {  
+            "name": "[variables('auth0ToApplicationInsightFunctionName')]",
+            "type": "Microsoft.Web/sites",
+            "apiVersion": "2022-03-01",
+            "location": "[variables('location')]",
+            "kind": "functionapp",
+            "identity": {
+                "type": "UserAssigned",
+                "userAssignedIdentities": { "[variables('identityId')]": {} }
+            },
+          "dependsOn": [
+            "[concat('Microsoft.KeyVault/vaults/', variables('shortName'), '-kv', '/secrets/ConnectionStrings--Auth0ToAI-Storage')]",
+            "[resourceId('Microsoft.Web/serverfarms', concat(parameters('uniquePrefixName'), '-', variables('projectName'), '-', 'ASP'))]",
+            "[variables('identityId')]"
+          ],
+          "properties": {
+            "reserved": false,
+            "name": "[variables('auth0ToApplicationInsightFunctionName')]",
+            "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', concat(parameters('uniquePrefixName'), '-', variables('projectName'), '-', 'ASP'))]",
+            "siteConfig": {
+              "numberOfWorkers": 1,
+              "http20Enabled": false,
+              "alwaysOn": false
+            },
+            "httpsOnly": true,
+
+            "keyVaultReferenceIdentity": "[variables('identityId')]"
+          },
+            "resources": [
+                {
+                    "name": "appsettings",
+                    "type": "config",
+                    "apiVersion": "2022-03-01",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Web/sites', variables('auth0ToApplicationInsightFunctionName'))]"
+                    ],
+                  "properties": {
+                    "APPINSIGHTS_INSTRUMENTATIONKEY": "[reference(resourceId('microsoft.insights/components', variables('applicationInsightsName')), '2020-02-02-preview').InstrumentationKey]",
+                    "APPLICATIONINSIGHTS_CONNECTION_STRING": "[reference(resourceId('microsoft.insights/components', variables('applicationInsightsName')), '2020-02-02-preview').ConnectionString]",
+                    "Auth0LogConnectionString": "[parameters('auth0LogConnectionString')]",
+                    "AzureWebJobsStorage__clientId": "[reference(resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', variables('identityName'))).clientId]",
+                    "AzureWebJobsStorage__accountName": "[variables('storageAccountName')]",
+                    "AzureWebJobsStorage__credential":"managedidentity",
+                    "FUNCTIONS_EXTENSION_VERSION": "~4",
+                    "FUNCTIONS_WORKER_RUNTIME": "dotnet",
```

Comment:
use dotnet-isolated which is the newer offer unless there is a reason not to. if there is a reason, leave a comment.